main/sdr.go

```diff
@@ -289,6 +300,92 @@ func (f *OGN) read() {
 	}
 }
 
+func (e *AIS) read() {
+	defer e.wg.Done()
+	log.Println("Entered AIS read() ...")
+	cmd := exec.Command(STRATUX_HOME + "/bin/rtl_ais", "-T", "-k", "-p", strconv.Itoa(e.ppm), "-d", strconv.Itoa(e.indexID))
+	stdout, _ := cmd.StdoutPipe()
+	stderr, _ := cmd.StderrPipe()
+
+	err := cmd.Start()
+	if err != nil {
+		log.Printf("Error executing " + STRATUX_HOME + "/bin/rtl_ais: %s\n", err)
+		// don't return immediately, use the proper shutdown procedure
+		shutdownES = true
+		for {
+			select {
+			case <-e.closeCh:
+				return
+			default:
+				time.Sleep(1 * time.Second)
+			}
+		}
+	}
+
+	log.Println("Executed " + cmd.String() + " successfully...")
+
+	done := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			case <-e.closeCh:
+				log.Println("AIS read(): shutdown msg received, calling cmd.Process.Kill() ...")
+				err := cmd.Process.Kill()
+				if err == nil {
+					log.Println("kill successful...")
+				}
+				return
+			default:
+				time.Sleep(1 * time.Second)
+			}
+		}
+	}()
+
+	stdoutBuf := make([]byte, 1024)
+	stderrBuf := make([]byte, 1024)
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			default:
+				n, err := stdout.Read(stdoutBuf)
+				if err == nil && n > 0 {
+					m := AISTermMessage{Text: string(stdoutBuf[:n]), Source: "stdout"}
+					logAISTermMessage(m)
+				}
+			}
+		}
+	}()
+
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			default:
+				n, err := stderr.Read(stderrBuf)
+				if err == nil && n > 0 {
+					m := AISTermMessage{Text: string(stderrBuf[:n]), Source: "stderr"}
+					logAISTermMessage(m)
+				}
+			}
+		}
+	}()
+
+	cmd.Wait()
+
+	// we get here if A) the dump1090 process died
```

Comment:
Don't kill dump1090 please :(

Reply:
nope... I set it as 162Mhz, which is the center freq.